diff --git a/dist/config.mjs b/dist/config.mjs
index e79266c..ffe9f93 100644
--- a/dist/config.mjs
+++ b/dist/config.mjs
@@ -318,6 +318,18 @@ async function resolveConfig(config) {
   );
   resolvedConfig.plugins.push(NuxtVitestEnvironmentOptionsPlugin(resolvedConfig.test.environmentOptions));
   if (resolvedConfig.test.browser?.enabled) {
+    resolvedConfig.plugins.push({
+      name: "nuxt:test-utils:browser-client-environment",
+      configEnvironment(name) {
+        if (name === "client") {
+          return {
+            dev: {
+              moduleRunnerTransform: false
+            }
+          };
+        }
+      }
+    });
     if (resolvedConfig.test.environment === "nuxt") {
       resolvedConfig.test.setupFiles = Array.isArray(resolvedConfig.test.setupFiles) ? resolvedConfig.test.setupFiles : [resolvedConfig.test.setupFiles].filter(Boolean);
       const resolver = createResolver(import.meta.url);
