diff --git a/dist/api.d.mts b/dist/api.d.mts
index fb1b8f99d9814b38bb5226be70994c606389ae0d..3277157438d6d39ce23571b4433aeb913e210a3d 100644
--- a/dist/api.d.mts
+++ b/dist/api.d.mts
@@ -26,7 +26,7 @@ declare const generateSchema: (opts: {
   adapter: DBAdapter$1;
   file?: string;
   options: BetterAuthOptions;
-}) => Promise<SchemaGeneratorResult> | Promise<{
+}) => Promise<SchemaGeneratorResult | {
   code: string;
   fileName: string;
   overwrite: boolean | undefined;
diff --git a/dist/api.mjs b/dist/api.mjs
index 4ce4a6ab6be5cacff7a903512a2d36f0d34a3ffa..5c1895922e9744d8ab3ab81e5c8044d4023de550 100644
--- a/dist/api.mjs
+++ b/dist/api.mjs
@@ -26,6 +26,10 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 		schema: tables,
 		usePlural: adapter.options?.adapterConfig?.usePlural
 	});
+	const getSingularModelName = initGetModelName({
+		schema: tables,
+		usePlural: false
+	});
 	const getFieldName = initGetFieldName({
 		schema: tables,
 		usePlural: adapter.options?.adapterConfig?.usePlural
@@ -51,9 +55,9 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 			}
 			const type = field.type;
 			if (typeof type !== "string") if (Array.isArray(type) && type.every((x) => typeof x === "string")) return {
-				sqlite: `text({ enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
+				sqlite: `text('${name}', { enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
 				pg: `text('${name}', { enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
-				mysql: `mysqlEnum([${type.map((x) => `'${x}'`).join(", ")}])`
+				mysql: `mysqlEnum('${name}', [${type.map((x) => `'${x}'`).join(", ")}])`
 			}[databaseType];
 			else throw new TypeError(`Invalid field type for field ${name} in model ${modelName}`);
 			const dbTypeMap = {
@@ -155,7 +159,7 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 		const foreignFields = Object.entries(table.fields).filter(([_, field]) => field.references);
 		for (const [fieldName, field] of foreignFields) {
 			const referencedModel = field.references.model;
-			const relationKey = getModelName(referencedModel);
+			const relationKey = getSingularModelName(referencedModel);
 			const fieldRef = `${getModelName(tableKey)}.${getFieldName({
 				model: tableKey,
 				field: fieldName
@@ -564,16 +568,16 @@ const adapters = {
 	drizzle: generateDrizzleSchema,
 	kysely: generateKyselySchema
 };
-const generateSchema = (opts) => {
+const generateSchema = async (opts) => {
 	const adapter = opts.adapter;
-	const generator = adapter.id in adapters ? adapters[adapter.id] : null;
-	if (generator) return generator(opts);
 	if (adapter.createSchema) return adapter.createSchema(opts.options, opts.file).then(({ code, path: fileName, overwrite }) => ({
 		code,
 		fileName,
 		overwrite
 	}));
-	throw new Error(`${adapter.id} is not supported. If it is a custom adapter, please request the maintainer to implement createSchema`);
+	const generator = adapters[adapter.id] ?? null;
+	if (generator) return await generator(opts);
+	throw new Error(`${adapter.id} is not supported. If it is a custom adapter, please request the maintainer to implement the "createSchema" method on the adapter.`);
 };
 //#endregion
 export { adapters, generateDrizzleSchema, generateKyselySchema, generatePrismaSchema, generateSchema };
diff --git a/dist/index.mjs b/dist/index.mjs
index 1c30d132ba9238f605176cc5a5c37f7e0284a119..62ca14bc56a72a0ec39ba0c75d61ec6d57747dec 100755
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -596,6 +596,10 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 		schema: tables,
 		usePlural: adapter.options?.adapterConfig?.usePlural
 	});
+	const getSingularModelName = initGetModelName({
+		schema: tables,
+		usePlural: false
+	});
 	const getFieldName = initGetFieldName({
 		schema: tables,
 		usePlural: adapter.options?.adapterConfig?.usePlural
@@ -621,9 +625,9 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 			}
 			const type = field.type;
 			if (typeof type !== "string") if (Array.isArray(type) && type.every((x) => typeof x === "string")) return {
-				sqlite: `text({ enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
+				sqlite: `text('${name}', { enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
 				pg: `text('${name}', { enum: [${type.map((x) => `'${x}'`).join(", ")}] })`,
-				mysql: `mysqlEnum([${type.map((x) => `'${x}'`).join(", ")}])`
+				mysql: `mysqlEnum('${name}', [${type.map((x) => `'${x}'`).join(", ")}])`
 			}[databaseType];
 			else throw new TypeError(`Invalid field type for field ${name} in model ${modelName}`);
 			const dbTypeMap = {
@@ -725,7 +729,7 @@ const generateDrizzleSchema = async ({ options, file, adapter }) => {
 		const foreignFields = Object.entries(table.fields).filter(([_, field]) => field.references);
 		for (const [fieldName, field] of foreignFields) {
 			const referencedModel = field.references.model;
-			const relationKey = getModelName(referencedModel);
+			const relationKey = getSingularModelName(referencedModel);
 			const fieldRef = `${getModelName(tableKey)}.${getFieldName({
 				model: tableKey,
 				field: fieldName
@@ -1219,16 +1223,16 @@ const adapters = {
 	drizzle: generateDrizzleSchema,
 	kysely: generateKyselySchema
 };
-const generateSchema = (opts) => {
+const generateSchema = async (opts) => {
 	const adapter = opts.adapter;
-	const generator = adapter.id in adapters ? adapters[adapter.id] : null;
-	if (generator) return generator(opts);
 	if (adapter.createSchema) return adapter.createSchema(opts.options, opts.file).then(({ code, path: fileName, overwrite }) => ({
 		code,
 		fileName,
 		overwrite
 	}));
-	throw new Error(`${adapter.id} is not supported. If it is a custom adapter, please request the maintainer to implement createSchema`);
+	const generator = adapters[adapter.id] ?? null;
+	if (generator) return await generator(opts);
+	throw new Error(`${adapter.id} is not supported. If it is a custom adapter, please request the maintainer to implement the "createSchema" method on the adapter.`);
 };
 //#endregion
 //#region src/utils/add-cloudflare-modules.ts
